### PR TITLE
[Chore] Update Swift 3.0

### PR DESCRIPTION
**Pull request by quentinR**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,37 +175,40 @@ Yoshi.show()
...
 
-Finally, to invoke Yoshi in response to a 3D touch event, override and forward `touchesMoved:withEvent:` function with `minimumForcePercent` as an optional parameter.
+* Finally, to invoke Yoshi in response to a 3D touch event, add the `forceTouch` option in the `setupDebugMenu` invocations parameter as follows.
 
 ```swift
-override func touchesMoved(touches: Set<UITouch>, withEvent event: UIEvent?) {
-    Yoshi.touchesMoved(touches, withEvent: event)
-}
-````
+Yoshi.setupDebugMenu([/* YoshiMenu items */], invocations: [forceTouch])
```
Comment (htinlinn) [inline README.md]: missing `.` in front of `forceTouch`

File: README.md [hunk 2]
```diff
@@ -167,37 +175,40 @@ Yoshi.show()
...
 ```
 
-To invoke Yoshi in in response to a multi-touch event, override and forward `touchesBegan:withEvent:` function. Optionally, the minimum number of touches required for invocation can be specified.
+* To invoke Yoshi in in response to a multi-touch event, add the `multiTouch` option in the `setupDebugMenu` invocations parameter as follows.
 
 ```swift
-override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
-    Yoshi.touchesBegan(touches, withEvent: event)
-}
+Yoshi.setupDebugMenu([/* YoshiMenu items */], invocations: [multiTouch])
```
Comment (htinlinn) [inline README.md]: missing `.` in front of `multiTouch`

File: README.md [hunk 3]
```diff
@@ -167,37 +175,40 @@ Yoshi.show()
...
+
+To specify which option you want exacly you just need to pass the ones you want to the `setupDebugMenu` function like this:
+
+* To invoke Yoshi in response to a shake-motion gesture, add the `shakeMotionGesture` option in the `setupDebugMenu` invocations parameter as follows.
 
 ```swift
-override func motionBegan(motion: UIEventSubtype, withEvent event: UIEvent?) {
-    Yoshi.motionBegan(motion, withEvent: event)
-}
+Yoshi.setupDebugMenu([/* YoshiMenu items */], invocations: [shakeMotionGesture])
```
Comment (htinlinn) [inline README.md]: missing `.` in front of `shakeMotionGesture`

File: README.md [hunk 4]
```diff
@@ -167,37 +175,40 @@ Yoshi.show()
...
 Yoshi.show()
 ```
 
-In addition to the vanilla invocation option, Yoshi can also be invoked in response to motion or touch events. To do this, simply forward the motion and touch-related `UIResponder` events to their corresponding Yoshi event-handling functions.
+In addition to the vanilla invocation option, Yoshi can also be invoked in response of 3 different options of motion or touch events.   
+If you want to enable all of those 3 following options you can simply pass the `all` option to the `setupDebugMenu`, although this option is already the default one.
 
-To invoke Yoshi in response to a shake-motion gesture, override and forward `motionBegan:withEvent:` function as follows.
+```swift
+Yoshi.setupDebugMenu([/* YoshiMenu items */], invocations: [all])
```
Comment (htinlinn) [inline README.md]: missing `.` in front of `all`

File: README.md [hunk 5]
```diff
@@ -167,37 +175,40 @@ Yoshi.show()
 
 ### Invocation Options
 
-Yoshi can be invoked a number of different ways. The simplest way is to manually invoke using the `show()` function.
+Yoshi can be invoked a number of different options. The simplest way is to manually invoke using the `show()` function.
```
Comment (htinlinn) [inline README.md]: Add the word _with_ before _a number of different options_ here?

File: README.md [hunk 6]
```diff
@@ -167,37 +175,40 @@ Yoshi.show()
...
+If you want to enable all of those 3 following options you can simply pass the `all` option to the `setupDebugMenu`, although this option is already the default one.
 
-To invoke Yoshi in response to a shake-motion gesture, override and forward `motionBegan:withEvent:` function as follows.
+```swift
+Yoshi.setupDebugMenu([/* YoshiMenu items */], invocations: [all])
+/// Or simply
+Yoshi.setupDebugMenu([/* YoshiMenu items */])
+```
+
+To specify which option you want exacly you just need to pass the ones you want to the `setupDebugMenu` function like this:
```
Comment (htinlinn) [inline README.md]: Typo here. _exactly_.

File: README.md (hunk 5 above)
Comment (quentinR) [inline README.md]: done

File: README.md (hunk 4 above)
Comment (quentinR) [inline README.md]: done

File: README.md (hunk 6 above)
Comment (quentinR) [inline README.md]: done

File: README.md (hunk 3 above)
Comment (quentinR) [inline README.md]: done

File: README.md (hunk 2 above)
Comment (quentinR) [inline README.md]: done

File: Yoshi.podspec [hunk 7]
```diff
@@ -8,7 +8,7 @@
 
 Pod::Spec.new do |s|
   s.name             = "Yoshi"
-  s.version          = "1.1.0"
+  s.version          = "1.2.0"
```
Comment (MichaelCampbell) [inline Yoshi.podspec]: maybe this swift 3 change should be version 2.0 instead. That way if there are issues that need to be resolved in the 1.x.y line, they can be fixed.

Comment (quentinR) [inline Yoshi.podspec]: The fix of the bug gesture was 1.1.1, so I figure we could fix 2.3 on 1.1.x

File: Yoshi/Yoshi/Yoshi.swift [hunk 8]
```diff
@@ -45,8 +45,9 @@ public final class Yoshi {
      - parameter event:                   (UIEvent) the event captured by the original touchesBegan call
      - parameter minimumTouchRequirement: (Int) the minimum number of touches required to show the debug menu.
      */
-    public class func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?, minimumTouchRequirement: Int = 3) {
-        guard event?.allTouches()?.count >= minimumTouchRequirement
+    public class func touchesBegan(_ touches: Set<UITouch>, withEvent event: UIEvent?,
+                                   minimumTouchRequirement: Int = 3) {
+        guard (event?.allTouches?.count)! >= minimumTouchRequirement
```
Comment (MichaelCampbell) [inline Yoshi/Yoshi/Yoshi.swift]: what happens if `event` or `allTouches` is nil here? maybe use a let instead?

File: Yoshi.podspec (hunk 7 above)
Comment (quentinR) [inline Yoshi.podspec]: Cool beans

File: Yoshi/Yoshi/Yoshi.swift (hunk 8 above)
Comment (quentinR) [inline Yoshi/Yoshi/Yoshi.swift]: I added some logic in the UIWindows extension to be able to change this.

File: Yoshi.podspec [hunk 9]
```diff
@@ -8,7 +8,7 @@
 
 Pod::Spec.new do |s|
   s.name             = "Yoshi"
-  s.version          = "1.1.0"
+  s.version          = "2.0"
```
Comment (MichaelCampbell) [inline Yoshi.podspec]: 2.0.0

Comment (quentinR) [inline Yoshi.podspec]: I don't think it's useful

Comment (MichaelCampbell) [inline Yoshi.podspec]: it's convention, take a look here https://github.com/Alamofire/AlamofireImage/commit/33c66155f7990890c730b5847909e12fbae93842#diff-3c97d208da64ea8b4acd0306e8caf4adL3

Comment (htinlinn) [inline Yoshi.podspec]: I would agree with @MichaelCampbell here. We should go with 2.0.0.

Comment (quentinR) [inline Yoshi.podspec]: Alright let's update that then!

Comment (quentinR) [inline Yoshi.podspec]: done

**Pull request merged**